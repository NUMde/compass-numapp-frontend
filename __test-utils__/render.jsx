/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { render as rtlRender } from '@testing-library/react-native';
import { combineReducers, createStore } from 'redux';
import { Provider } from 'react-redux';

import UserReducer from '../src/store/user.slice';
import GlobalsReducer from '../src/store/globals.slice';
import QuestionnaireReducer from '../src/store/questionnaire.slice';

const testReducer = combineReducers({
  User: UserReducer,
  Globals: GlobalsReducer,
  Questionnaire: QuestionnaireReducer,
});

function renderWithRedux(
  ui,
  {
    initialState = {
      User: UserReducer.initialState,
      Globals: GlobalsReducer.initialState,
      Questionnaire: QuestionnaireReducer.initialState,
    },
    store = createStore(testReducer, initialState),
    ...renderOptions
  } = {},
) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>;
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

export * from '@testing-library/react-native';

export { renderWithRedux };
