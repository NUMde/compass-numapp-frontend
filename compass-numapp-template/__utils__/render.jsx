import React from 'react';
import { Provider } from 'react-redux';
// eslint-disable-next-line import/no-extraneous-dependencies
import { render } from '@testing-library/react-native';
import { buildStore } from './mockStore';

let store = null;

export function renderWithRedux(subComponent, state = {}) {
  if (!store) {
    store = buildStore(state);
  }

  const queries = render(<Provider store={store}>{subComponent}</Provider>);

  return {
    ...queries,
    store,
  };
}

export function initRenderModule(initialState = {}) {
  store = buildStore(initialState);

  return store;
}
