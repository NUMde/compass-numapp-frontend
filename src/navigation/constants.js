/**
 * enum for all route names
 */
const Routes = {
  LANDING: 'Landing',
  LOGIN: 'Login',
  CHECK_IN: 'CheckIn',
  SURVEY: 'Survey',
  ABOUT: 'About',
  WEBVIEW: 'Webview',
  LEGAL_INFORMATION: 'LegalInformation',
};

/**
 * enum for all stacks
 */
const Stacks = {
  SIGNED_IN: 'SignedIn',
  SIGNED_OUT: 'SignedOut',
};

export { Routes, Stacks };
