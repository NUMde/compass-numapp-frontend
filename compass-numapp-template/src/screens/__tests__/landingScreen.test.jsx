import React from 'react';

import { renderWithRedux, fireEvent } from '__test-utils__/render';

import en from '~CUSTOMIZATION/translations/en';

import { Stacks, Routes } from '~navigation/constants';
import LandingScreen from '../landingScreen';

describe('LandingScreen', () => {
  it('should render screen', () => {
    const navigate = jest.fn();
    const goBack = jest.fn();
    const { getByText, toJSON } = renderWithRedux(
      <LandingScreen navigation={{ navigate, goBack }} />,
    );
    expect(toJSON()).toMatchSnapshot();
    expect(getByText(en.login.landing.buttonText)).toBeTruthy();
  });

  it('should navigate to loginScreen when pressing button', () => {
    const navigate = jest.fn();
    const goBack = jest.fn();
    const { getByText } = renderWithRedux(
      <LandingScreen navigation={{ navigate, goBack }} />,
    );

    fireEvent.press(getByText(en.login.landing.buttonText));
    expect(navigate).toHaveBeenCalledWith(Routes.LOGIN);
  });

  it('should navigate to checkInScreen when subjectId is present', () => {
    const navigate = jest.fn();
    const goBack = jest.fn();
    renderWithRedux(<LandingScreen navigation={{ navigate, goBack }} />, {
      initialState: { User: { subjectId: 'someId' } },
    });

    expect(navigate).toHaveBeenCalledWith(Stacks.SIGNED_IN, {
      screen: Routes.CHECK_IN,
    });
  });
});
