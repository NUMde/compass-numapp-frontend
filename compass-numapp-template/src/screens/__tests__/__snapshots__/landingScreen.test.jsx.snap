// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingScreen should render screen 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F8F9FA",
      "flexDirection": "column",
      "height": "100%",
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "backgroundColor": "#F8F9FA",
        "flexDirection": "column",
        "height": 354.06976744186045,
        "paddingTop": 30,
        "width": "100%",
      }
    }
  >
    <Image
      resizeMode="cover"
      source={
        Object {
          "process": [Function],
        }
      }
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": 354.06976744186045,
            "width": "100%",
          },
          undefined,
        ]
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginVertical": 20,
          "width": "90%",
        }
      }
    >
      <View
        style={
          Object {
            "height": 26,
            "width": 26,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
        }
      }
    >
      <Image
        resizeMode="contain"
        source={
          Object {
            "process": [Function],
          }
        }
        style={
          Object {
            "alignSelf": "center",
            "bottom": 18.313953488372093,
            "maxHeight": 432.29448350459705,
            "maxWidth": 650,
            "position": "relative",
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "backgroundColor": "#F8F9FA",
          "flexDirection": "column",
          "height": "100%",
        },
      ]
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
          "width": "100%",
        }
      }
    >
      <RCTScrollView
        contentContainerStyle={
          Object {
            "flexGrow": 1,
          }
        }
        enableOnAndroid={true}
        onScroll={[Function]}
        scrollEventThrottle={16}
        style={
          Object {
            "flex": 1,
            "flexDirection": "column",
            "width": "100%",
          }
        }
      >
        <View>
          <View
            style={
              Object {
                "backgroundColor": "#F8F9FA",
                "flexDirection": "column",
                "height": "100%",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "marginBottom": 61.04651162790697,
                  "paddingTop": 61.04651162790697,
                }
              }
            >
              <Text
                style={
                  Object {
                    "alignSelf": "center",
                    "fontFamily": "IBMPlexSans-SemiBold",
                    "fontSize": 33.75,
                    "lineHeight": 43.125,
                    "textAlign": "center",
                    "width": "80%",
                  }
                }
              >
                Welcome to COMPASS
(React Native Client)
              </Text>
              <Text
                style={
                  Object {
                    "alignSelf": "center",
                    "color": "#1F2532",
                    "fontFamily": "IBMPlexSans",
                    "fontSize": 28.125,
                    "lineHeight": 37.5,
                    "marginHorizontal": 69.76744186046511,
                    "marginVertical": 34.883720930232556,
                    "textAlign": "center",
                  }
                }
              >
                This is the Landing Screen.

It can be used to display a welcome message, detailing the following login-process, explaining the handling of personal information, etc.

The logo file (as well as the logoBackground) can be updated by updating the 'CUSTOMIZATION' folder.
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                  "height": "100%",
                  "justifyContent": "flex-end",
                  "marginBottom": 36,
                  "marginTop": 20,
                }
              }
            >
              <View
                accessibilityHint="Starts login process and redirects to QRCode scanner."
                accessibilityLabel="Navigate to Login Screen"
                accessibilityRole="button"
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignSelf": "center",
                    "backgroundColor": "#1F2532",
                    "borderColor": "#1F2532",
                    "borderRadius": 50,
                    "borderWidth": 0,
                    "bottom": 0,
                    "opacity": 1,
                    "padding": 15,
                    "paddingHorizontal": 26.162790697674417,
                    "paddingVertical": 26.162790697674417,
                    "width": "80%",
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "alignSelf": "center",
                      "color": "#F8F9FA",
                      "fontFamily": "IBMPlexSans-SemiBold",
                      "fontSize": 26.25,
                      "lineHeight": 35.625,
                      "textAlign": "center",
                    }
                  }
                >
                  Navigate to Login Screen
                </Text>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
</View>
`;
